### PR TITLE
enhance switchdiscover to deal with finding nodes hash

### PR DESCRIPTION
**Pull request by zet809**
The issue to enhance the processing logic for the finding nodes hash in #3953 .

The modification:
1. the nodes hash is using 'mac' as the keys, so no need to compare whether this is multiple entries for same mac address.
2. for difference MAC with same IP, the original process is merge verdor together and create a new entry. No special process in this modification.
No modification from outside.

### CODE REVIEW

File: xCAT-server/lib/xcat/plugins/switchdiscover.pm [hunk 1]
```diff
@@ -339,74 +339,28 @@ sub process_request {
...
-                        delete $result->{$new_mac};
-                    }
-                }
-                if ( $old_name && ($old_name eq $new_name)) {
-                    #appending mac address to end of hostname
-                    my $mac_str = lc($old_mac);
+            foreach (keys %$tmp_result) {
+                $result->{$_} = $tmp_result->{$_};
+                #appending mac address to end of hostname
+                my $name = $result->{$_}->{name};
```
Comment (cxhong) [inline xCAT-server/lib/xcat/plugins/switchdiscover.pm]: sometime, nmap scan return empty hostname, need to check if this $name defined.

**Comment by cxhong** [comment]
@zet809 , I verified on one of system, here are the results:
output from switchdiscover command with nmap scan,  the nmap didn't show hostname, so the output of hostname will show "-$mac",  I think we should add "switch-$mac" or other default hostname
```
[mgt03]: ip             name                    vendor                                                  mac
[mgt03]: ------------   ------------            ------------                                            ------------
[mgt03]: 192.168.74.200 c685f2u24vc             Edgecore Networks Switch                                8C:EA:1B:E8:70:00
[mgt03]: 192.168.74.202 -8cea1be872c0           Edgecore Networks Switch                                8C:EA:1B:E8:72:C0
[mgt03]: 192.168.74.204 -8cea1be874c0           Edgecore Networks Switch                                8C:EA:1B:E8:74:C0
````
the pdu discover looks good
```
[mgt03]: ip             name                    vendor                                                  mac
[mgt03]: ------------   ------------            ------------                                            ------------
[mgt03]: 192.168.74.23  coral-pdu-98be947f006f  Linux coral-pdu 3.14.52+g5f6f0a5 #1 SMP PREEMPT Fri Nov 18 14:55:31 CST 2016 armv7l  98:be:94:7f:00:6f
[mgt03]: 192.168.74.24  coral-pdu               Linux coral-pdu 3.14.52+g5f6f0a5 #1 SMP PREEMPT Fri Nov 18 14:55:31 CST 2016 armv7l  98:be:94:7f:00:77
[mgt03]: 192.168.74.62  coral-pdu-98be947f006e  Linux coral-pdu 3.14.52+g5f6f0a5 #1 SMP PREEMPT Fri Nov 18 14:55:31 CST 2016 armv7l  98:be:94:7f:00:6e
````

**Comment by zet809** [comment]
hi, @cxhong , modified, pls help to review. The output looks like this now:
```
Discovering switches using nmap for 192.168.74.200-207. It may take long time...
ip          	name                	vendor                                            	mac
------------	------------        	------------                                      	------------
192.168.74.200	c685f2u24vc         	Edgecore Networks Switch                          	8C:EA:1B:E8:70:00
192.168.74.202	switch-8cea1be872c0 	Edgecore Networks Switch                          	8C:EA:1B:E8:72:C0
192.168.74.204	switch-8cea1be874c0 	Edgecore Networks Switch                          	8C:EA:1B:E8:74:C0
```

**Comment by cxhong** [comment]
Thanks @zet809 .  I will merge it

**Pull request merged**